### DISCUSSION TITLE
Alpine Images

### DISCUSSION THREAD

**Discussion by Milo123459** [railwayapp/nixpacks#825] [RFCs]
Alpine Linux is a distribution of Linux that is secure and small. I think we should use it over Ubuntu/Debian whenever possible, which is  basically when a plan doesn't require any apt packages. This would make builds smaller. Related: #729

**Comment by coffee-cup** [comment]
Alpine won't work in a lot of cases since it doesn't support glibc. You instead need to compile with musl. This can break packages/builds that are not statically linked. We try to statically link Rust binaries, but there are some cases where it is not possible (https://github.com/railwayapp/nixpacks/blob/main/src/providers/rust.rs#L74). I'd imagine that there are a bunch of issues with python/ruby/node libs that compile C extensions or run dynamically linked binaries.

Having the option to use Alpine as the base image is a good idea. But I don't think it should be the default.

**Comment by thrzl** [comment]
distroless might be a better alternative, i'm aware that [google maintains some](https://github.com/GoogleContainerTools/distroless) distroless images.
